hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -463,17 +575,18 @@ static inline void setup8ByteWrite(uint64_t addr, const void *data)
 #endif /* ifdef USING_KSDK2 */
 }
 
+__RAMFUNC
 static inline void setupProgramSection(uint64_t addr, const void *data, size_t cnt)
 {
 #ifdef USING_KSDK2
     static const uintptr_t FlexRAMBase = FSL_FEATURE_FLASH_FLEX_RAM_START_ADDRESS;
-    memcpy((void *)FlexRAMBase, (const uint8_t *)data, cnt);
+    MEMCPY((void *)FlexRAMBase, (const uint8_t *)data, cnt);
```

Comment:
If the entire launchCommandAndWaitForCompletion() function is in a ram array then it would work.  KSDK2 driver does something similar by copying a C function into ram a ram buffer with a memcpy https://github.com/mbedmicro/mbed/blob/master/hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/drivers/fsl_flash.c#L1946.  I have discussed this with @flit at NXP and last I heard the plan was to write this function in assembly as @kjbracey-arm has done so it is more portable and won't change based on C optimization settings.

Also as @kjbracey-arm mentioned, the flash cache should be cleared after an erase or write to prevent old data from being used.  The KSDK2 code to do this can be found here https://github.com/mbedmicro/mbed/blob/master/hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/drivers/fsl_flash.c#L2025